src/Juvix/Core/Parameterisations/Unit.hs

```diff
@@ -0,0 +1,50 @@
+module Juvix.Core.Parameterisations.Unit where
+
+import Juvix.Core.Types hiding
+  ( apply,
+    parseTy,
+    parseVal,
+    reservedNames,
+    reservedOpNames,
+    typeOf,
+  )
+import Juvix.Library hiding ((<|>))
+import Text.ParserCombinators.Parsec
+import qualified Text.ParserCombinators.Parsec.Token as Token
+import Prelude (String)
+
+-- k: primitive type: unit
+data UnitTy
+  = TUnit
+  deriving (Show, Eq)
+
+-- c: primitive constant and f: functions
+data UnitVal
+  = Unit
+  deriving (Show, Eq)
+
+typeOf ∷ ∀ a. ([UnitTy] → a) → UnitVal → Either a UnitTy
+typeOf _ Unit = Right TUnit
+
+apply ∷ UnitVal → UnitVal → Maybe UnitVal
+apply _ _ = Nothing
+
+parseTy ∷ Token.GenTokenParser String () Identity → Parser UnitTy
+parseTy lexer = do
+  Token.reserved lexer "TUnit"
```

Comment:
`()` is the common shorthand, it's not a big deal though.

Reply:
They can't both be ```()``` so I've changed the type one to be "Unit" and the value to be "()". But the value one's test doesn't pass atm. Feel free to fix it and merge or I can fix it tomorrow.

Reply:
Fixed it, was a very minor change in the order in which things are parsed.

See #154 for the exact details